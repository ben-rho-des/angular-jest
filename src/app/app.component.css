html,
body {
    height: 100%;
}

body {
    font-family: sans-serif;
    background-image: url(/assets/where.gif);
    background-size: 100px;
    margin: 0;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.containter {
    width: 600px;
    background: #eee;
    border: 1px solid #d2d2d2;
    border-radius:6px;
    padding: 2rem;
    flex: 1;
}

button {
    background: #00cc00;
    border-radius: 3px;
    border: 0;
    cursor: pointer;
    font-weight: bold;
    font-size: 100%;
    padding: 15px 30px;
}

button:hover,
button:active {
    background: #009900;
}

h1 {
    margin-top: 0;
}
