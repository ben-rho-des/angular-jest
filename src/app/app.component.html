<div class="containter">
  <h1>silly app</h1>
  <div *ngIf="step === 1;else steptwocontent">
    <app-stepone (valid)="onValid($event)"></app-stepone>
  </div>
</div>

<ng-template #steptwocontent>
  <app-steptwo></app-steptwo>
</ng-template>