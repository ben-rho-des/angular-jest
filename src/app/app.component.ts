import { Component, ViewEncapsulation } from '@angular/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css'],
  encapsulation: ViewEncapsulation.None
})
export class AppComponent {

  private step = 1;

  onValid(valid: boolean) {
    this.step = (!valid) ? 1 : 2;
  }
}
