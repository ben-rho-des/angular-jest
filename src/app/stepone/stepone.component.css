input[type='text'] {
    display: block;
    margin-bottom: 1rem;
    margin-top: .5rem;
    border-radius: 3px;
    padding: 5px;
    border: 1px solid #d2d2d2;
}

fieldset {
    border: 0;
    padding: 0;
    margin: 1rem 0;
}

legend {
    margin-bottom: 0.5rem;
}