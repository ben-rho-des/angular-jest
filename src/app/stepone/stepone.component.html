<form #f="ngForm" (ngSubmit)="submit(f.value)">
  <label for="name">Name of something: </label>
  <input type="text" id="name" name="name" required [(ngModel)]="model.name"/>
  <label for="type">Type of named thing:</label>
  <input type="text" id="type" name="type" requireds [(ngModel)]="model.type"/>
  <button type="submit">Submit</button>
</form>