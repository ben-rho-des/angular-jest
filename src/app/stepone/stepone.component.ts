import { Component, EventEmitter, Input, Output } from '@angular/core';
import { FormControl } from '@angular/forms';

@Component({
  selector: 'app-stepone',
  templateUrl: './stepone.component.html',
  styleUrls: ['./stepone.component.css']
})
export class SteponeComponent {

  @Output() valid = new EventEmitter<boolean>();
  model = {name: '', type: ''};

  constructor() { }

  public submit(value) {
    const valid = (value.name !== '' && value.type !== '');
    this.valid.emit(valid);
  }
}
