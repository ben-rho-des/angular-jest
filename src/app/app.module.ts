import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';

import { AppComponent } from './app.component';
import { SteponeComponent } from './stepone/stepone.component';
import { SteptwoComponent } from './steptwo/steptwo.component';


@NgModule({
  declarations: [
    AppComponent,
    SteponeComponent,
    SteptwoComponent
  ],
  imports: [
    BrowserModule,
    FormsModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
