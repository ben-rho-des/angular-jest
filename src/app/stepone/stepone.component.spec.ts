import { async, ComponentFixture, TestBed, getTestBed } from '@angular/core/testing';
import { ChangeDetectionStrategy, Component, Injector, DebugElement } from '@angular/core';
import { AppModule } from '../app.module';
import { SteponeComponent } from './stepone.component';

describe('SteponeComponent', () => {
  let component: SteponeComponent;
  let fixture: ComponentFixture<SteponeComponent>;

  setupTestBed({
    imports: [AppModule]
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(SteponeComponent);

    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snapshot should pass', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('do some other complete thing and compared snapshot', () => {
    const fakeInput = {name: 'Mr brown fox', type: 'He\'s quick and jumpy'};
    component.model = fakeInput;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
