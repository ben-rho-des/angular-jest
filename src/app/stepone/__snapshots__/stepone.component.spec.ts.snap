// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SteponeComponent do some other complete thing and compared snapshot 1`] = `
<app-stepone
  model={[Function Object]}
  valid={[Function EventEmitter]}
>
  <form
    class="ng-untouched ng-pristine ng-invalid"
    novalidate=""
  >
    <label
      for="name"
    >
      Name of something: 
    </label>
    <input
      class="ng-untouched ng-pristine ng-invalid"
      id="name"
      name="name"
      ng-reflect-model="Mr brown fox"
      ng-reflect-name="name"
      ng-reflect-required=""
      required=""
      type="text"
    />
    <label
      for="type"
    >
      Type of named thing:
    </label>
    <input
      class="ng-untouched ng-pristine ng-valid"
      id="type"
      name="type"
      ng-reflect-model="He's quick and jumpy"
      ng-reflect-name="type"
      requireds=""
      type="text"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</app-stepone>
`;

exports[`SteponeComponent snapshot should pass 1`] = `
<app-stepone
  model={[Function Object]}
  valid={[Function EventEmitter]}
>
  <form
    class="ng-untouched ng-pristine ng-valid"
    novalidate=""
  >
    <label
      for="name"
    >
      Name of something: 
    </label>
    <input
      class="ng-untouched ng-pristine ng-valid"
      id="name"
      name="name"
      ng-reflect-model=""
      ng-reflect-name="name"
      ng-reflect-required=""
      required=""
      type="text"
    />
    <label
      for="type"
    >
      Type of named thing:
    </label>
    <input
      class="ng-untouched ng-pristine ng-valid"
      id="type"
      name="type"
      ng-reflect-model=""
      ng-reflect-name="type"
      requireds=""
      type="text"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</app-stepone>
`;
